### PR TITLE
Check for valid phone numbers before subscribing

### LINKED ISSUE #630: [FEAT] Validate phone number for subscription
# ⚠️ IMPORTANT: Please fill out this template to give us as much information as possible to consider/implement the feature.

### Prerequisites

- [x] check this box if you have completed the following:
    - Reviewed the [contributing guidelines](https://github.com/COVID-19-electronic-health-system/.github/blob/master/CONTRIBUTING.md) and [support](https://github.com/COVID-19-electronic-health-system/.github/blob/master/SUPPORT.md) files
    - Reviewed the README file for the repository you are working in
    - Searched for relevant instructions [on our Discord server](https://discord.gg/pPERUuv)
    - Searched the issues of the repository you are working in to make sure one was not already filed

### Summary

Either disable the subscribe button if the phone number is not valid or show an error message once the subscribe button is pressed. Either way, don't make a request to the subscription service and inform the user of what went wrong.

- [x] Validate the phone number is of [E.164 format](https://www.twilio.com/docs/glossary/what-e164)
- [x] If the phone number is not valid, do not make a request to the subscription service
- [x] If the phone number is not valid, inform the user

### Motivation

Phone numbers being subscribed to the notification service should be of a consistent format and should be able to support international numbers.

### Possible Alternatives

Only do the validation on the backend. It would return a `400` status code and it could be handled on the frontend.

### Additional Context

[E.164 format](https://www.twilio.com/docs/glossary/what-e164)

### PR DESCRIPTION
**Pull request by SomeMoosery**
# ⚠️ __IMPORTANT: Please do not create a Pull Request without creating an issue first.__

All changes need to be discusssed before proceeding. Failure to do so may result in the pull request being rejected.

Please be sure to review our [Code of Conduct](https://github.com/COVID-19-electronic-health-system/.github/blob/master/CODE_OF_CONDUCT.md), [Contributing Guidelines](https://github.com/COVID-19-electronic-health-system/.github/blob/master/CONTRIBUTING.md), and [Support documentation](https://github.com/COVID-19-electronic-health-system/.github/blob/master/SUPPORT.md) before submitting a pull request.

-----

## Please include the issue number the pull request fixes by replacing YOUR-ISSUE-HERE in the text below.

Fixes #630 

## Summary

Checks phone numbers client-side to ensure they comply with E.164 which is necessary for Twilio.

## Details

- Checking on the frontend because this is a quick check to do and isn't worth a backend requests
- Updated success message to display a success snackbar similar to how failure messages display error snackbars

## Test Plan (required)

invalid-number

We don't have any tests for `More`, but probably should add some, especially since we're doing a lot of phone number checks. Will create an issue.

## Final Checklist

- [x] For CoronaTracker, did you bump the version in `package.json`?
    - Update the second decimal for a major change
    - Update the third decimal for a minor change
    - Numbers can go past 9, e.g. `1.0.9` => `1.0.10`
    - For more info, read about [Semantic Versioning](https://semver.org/)
- [x] Did you add any new tests as necessary?
- [x] Is your PR rebased off the most current master?
- [x] Have you squashed all commits? _(can be done at merge)_
- [x] Did you use `yarn` not `npm`? _(important!)_
- [x] Did you use Material-UI wherever possible?
- [x] Did you run `yarn lint` on the code?
- [x] Did you run all of your most recent changes locally to make sure everything is working?

### CODE REVIEW

File: client/src/components/More.jsx [hunk 1]
```diff
@@ -205,6 +218,9 @@ const More = () => {
             <DialogContentText className={classes.descriptionText}>
               Enter your phone number to subscribe/unsubscribe to the occasional question/survey to answer over text.
             </DialogContentText>
+            <DialogContentText className={classes.descriptionText}>
+              Please add your +country code before entering
+            </DialogContentText>
```
Comment (SomeMoosery) [inline client/src/components/More.jsx:223]: Does anyone have a better suggestion for nicely displaying in a non-engineer-y way of "Enter your phone number to match regex `/^\+[1-9]\d{1,14}$/` or `+<COUNTRY-CODE><PHONE-NUMBER>`?

**Comment by SomeMoosery** [comment]
My only guess as to why there's changes in `Table` here is that previous changes didn't lint them properly 😕

File: client/src/components/More.jsx (hunk 1 above)
Comment (acthelemann) [reply client/src/components/More.jsx:223]: Maybe show the image on [this page](https://support.twilio.com/hc/en-us/articles/223183008-Formatting-International-Phone-Numbers) or giving a link to a website that explains it more

Comment (SomeMoosery) [reply client/src/components/More.jsx:223]: Oh nice, didn't see that page - I'll add that in there 👍

**Pull request merged**